### DISCUSSION TITLE
심화프로젝트 3주차 팀회고

### DISCUSSION THREAD

**Discussion by yeji0512** [codeit-bootcamp-frontend/2-Weekly-Mission#444] [회고]
# 왜 회고를 해야하나요?
- **원활한 팀 프로젝트 진행을 위해, 회고가 좋은 도구가 되어줄거예요.** 
매주 팀 단위로 회고를 하며 어떤 점이 좋았고, 어떤 점이 아쉬웠고, 아쉬웠던 부분은 어떻게 개선해볼지 함께 이야기해보면 
그 다음주에는 더 나은 프로젝트를 진행할 수 있을거예요. 회고를 통해 함께 성장해보아요 🔥 
- **회고 시간만큼은 솔직한 생각, 마음을 터놓아보기! 😎** 
솔직한 회고를 통해 다음주는 더 즐겁게 프로젝트를 진행할 수 있을거예요.  

# 회고 방법
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/144666341/b1f1cc83-7d24-4156-910d-3fc9b6bb7555)
회고는 **“KPT 회고”** 방식으로 진행해볼거예요. 아래 방법대로 회고해주시면 됩니다. 
- 팀원들과 모여 그 주의 팀 프로젝트 활동을 돌아보며, '좋았던 점(Keep) / 아쉬웠던 점(Problem) / 시도해볼 것(Try)'으로 항목을 나누어 회고해주세요.
    - Keep: 현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분
    - Problem: 현재 우리팀이 가지고 있는 문제
    - Try: Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 (혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)
- [KPT 회고 피그마 템플릿](https://www.figma.com/file/0NHIhlM9vhYZDrtZa7nJn8/%ED%9A%8C%EA%B3%A0-%ED%85%9C%ED%94%8C%EB%A6%BF?type=whiteboard&node-id=0%3A1&t=p6vXDVEjQntHKaNP-1)을 활용해서, 팀 회고를 진행하셔도 됩니다. 단 템플릿 사용시, 팀별로 복제해서 사용해주세요. 꼭 템플릿을 사용하지 않아도 되니, 팀별로 편한 방식으로 회고를 진행해주세요. 
- **이번주 일요일까지, 팀 회고한 내용을 정리하여 댓글로 올려주세요.** 
(댓글에 예시 회고를 올려드릴테니, 참고해주세요. 피그마 템플릿으로 진행하셨다면 캡처본을 올려주셔도 돼요.)

# 참고 글
- [KPT 회고란 무엇인가?](https://brunch.co.kr/@jinha0802/35)
- [팀문화의 탄생](https://techblog.woowahan.com/2677/)

**Comment by hankim0904** [comment]
# 10팀😎
## Keep
- 코드 리뷰를 통해 컨벤션을 지키고 팀원들이 성장할 수 있는 피드백을 제공함
- 맡은 일을 성실하게 잘 하고 있음
- 아이디어를 적극적으로 제시함
- 활발한 토론을 통해 기획을 계속 발전시키고 있음
- 디자인, 백엔드, 프론트엔드가 서로 소통하기 위해 노력함
- 코드를 팀원들이 이해하기 쉽도록 PR을 상세하게 작성함

## Problem
- 협업 프로젝트가 처음이라 진행 속도에 대한 불안함이 있음

## Try
- PR 코드 리뷰 시 댓글을 달면 더 좋을 것 같음
- zustand와 react query를 빨리 사용하고 싶음

![39617016290825380](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/117327533/07a8d649-ddf1-4b0e-bcc3-1f220f01b0e2)

마지막 파트는 10팀이 점령하겠습니다+_+

**Comment by 9ummy9** [comment]
# 6팀

## Keep 👌
   - 각 페이지의 우선 순위를 나눠 디자인 파트와 협업이 잘 이루어졌다.
   - 백엔드, 디자인 파트, 팀원간의 소통이 잘 되고 있다.
   - 회의를 모두 성실히 참여하고 있다.
   - 스프린트 별 계획을 열심히 세운다.

## Problem 😢
   - 기획안에 수정할 부분이 남아있었다.
  
## Try 👍
   - 프로젝트 기획 시 세부적인 부분들을 더 꼼꼼히 작성한다.

<br>

![655a78a8a07a9a5f91d0aa00278a901a](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148167964/ef9fdcff-8d77-48ac-ad14-f8f693eaf574)

FE, BE, D까지 모든 팀원이 열심히, 성실히 참여하고 있어서 좋습니다😊👍👍👍
남은 기간도 파이팅!!


---

이번주부터 회고를 다같이 진행하기로 해서 FE, BE, D 모두 참여한 회고도 올립니다!
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148167964/9b2f3720-633c-41ad-8070-da5ab8d1e1a7)

**Comment by bjiwon324** [comment]
# 5팀 

## Keep
- 매일 회의록 작성 중 입니다. 
-  소통을 활발하게 합니다.
-  문제 해결을 위한 협력이 활발합니다.


## Problem
- 디테일한 컨벤션들을 정했어야했습니다.
-  코드 리뷰가 적극적이지 못한 것 같습니다.

##Try
-  적극적으로 코드 리뷰 하기
- 마지막까지 화이팅!!


![love-celebrate](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148836373/daef69d5-dacb-438a-9058-706ea037795e)

**Comment by CheeseB** [comment]
# 3팀 🦸🏼

![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/43297823/847a3a71-c83a-453e-b5d4-b09005c4f107)

**Comment by MoonPillGyeom** [comment]
## 8팀
## Keep
- 회의시간 잘 지킵니다.
- 소통이 활발해요 
- 각자 역할을 열심히 하고 매일 어제 한일, 오늘 한일을 공유하고 있습니다. 
- 각자 진행상황 공유가 잘 되고 있습니다
- R&R분배가 잘 되어 있는것 같습니다.
## Problem
- 코드리뷰가 활성화 되지 않아 서로의 코드에 대한 이해가 부족한 것 같습니다.
- pr 확인이 점점 늦어지는 것 같습니다.
- 각자 개인의 할일이 바빠 PR확인이 제때 이뤄지지 않는것 같습니다.
## Try
- 코드 리뷰 더 활성화...?,
- 인당 PR에 코멘트 1일 1회 정도 의무화...? 아니면 저번에 지선님이 제시하신 의견 등 리뷰 활성화 방안이 있어도 괜찮을 것 같습니다
- 각자 하루에 두 번 정도는 팀 레포 PR 확인하기로 해서 12시간 이내에는 머지될 수 있도록 하면 좋을 것 같습니다.
- 짧은 코멘트라도 각자 1PR 1리뷰 단다고 생각하면 좋을 것 같습니다.
- 깃허브 앱 다운 받아서 알림 활성화 해놓으면 팀 레포에 새롭게 올라오는 PR이나 코멘트를 놓치는 일이 줄어들 것 같습니다.
- 팀회의 시간에 안 되어있는 PR이 있다면 해결해 보는것도 좋을것 같습니다.

**Comment by rladngus133** [comment]
# 2팀

### Keep
팀 내 분위기가 좋고, 서로 맡은 일을 잘 진행하고 있어서 좋다.
팀원들이 이해하기 쉽게 PR에 상세하게 작성해 주고 있어서 좋다.
컨벤션 수정에 대한 의견과 정보 공유가 활발해서 좋다.

### Problem
개발 및 코드 리뷰 속도가 조금 더 빨라지면 좋을거 같다.
팀원이 작성한 공통 컴포넌트를 사용하기 어렵다.

### Try
하루 1시간 씩 코드 리뷰를 진행하는 시간을 가지고 싶다.

![image](https://blog.kakaocdn.net/dn/cHNtn1/btqJF0P3rog/Y03caEPvCzcSgZfVnl9p61/img.gif)

**Comment by SeanKim-416** [comment]
# 11팀

![스크린샷 2024-03-18 오전 10 08 04](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/56223156/19f51851-1d3f-472f-837a-5d0369091172)
<hr>

![스크린샷 2024-03-18 오전 10 08 13](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/56223156/4217e922-08ba-4a26-9a90-1f917b6c204e)
<hr>

![스크린샷 2024-03-18 오전 10 08 29](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/56223156/76dd390f-f662-4916-9840-e2a14cd34e32)

**Comment by muyahho** [comment]
4팀 회고
Keep

코드 작업 중 어려움이나 문제 발생 시 적극적으로 공유하고 협력하여 해결
회의 참여도 및 R&R 분배 적절
다양한 의견 제시를 통한 코드 개선
서로 배려하는 건강한 프로젝트 진행 분위기
Problem

코드 리뷰 활성화 부족
소통 명확성 개선 필요
코딩 실력 향상 및 문제 해결 능력 증진
PR 단위 조정 및 코드 리뷰 효율 개선
복잡한 에러 해결 능력 향상
Try

코드 리뷰 시 질문 던져 적극 참여 유도
PR 단위 축소 및 자주 올리기
적극적인 소통을 통한 문제 해결 및 정보 공유
보안성 등 깊이 있는 코드 구현 노력

**Comment by juncastle97** [comment]
# 9팀

## Keep
- 저번주보다 진행상황도 빨라지고 맡은 부분을 열심히 하고 있습니다.
- pr과 이슈를 세분화 해서 변화된 부분을 쉽게 파악할 수 있습니다.
- pr을 머지하기 전에 팀원들의 approve를 받아 피드백을 수용하여 작업합니다.

## Problem
- 이슈와 pr이 많아지는 만큼 확인을 하지 않으면 어떤 기능들이 추가되었는지 파악하기 어렵습니다.
- 모든 디자인과 기능이 나온 상태가 아니라 한번에 작업하기 힘듭니다.

## Try
- 다른 팀원의 pr을 잘 확인하자
- 이미 나온 디자인과 기능을 조금 더 유심히 살펴보자
